### PR TITLE
feat: support tree shaking for ES module

### PR DESCRIPTION
**Pull request by JounQin**

### CODE REVIEW

File: esm.mjs [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+      : (src) => {
+          if (src.match(/[^\x00-\xFF]/)) throw new RangeError(
+              'The string contains invalid characters.'
+          );
+          return fromUint8Array(
+              Uint8Array.from(src,c => c.charCodeAt(0))
+          );
+      };
+/**
+ * @deprecated since 3.0.0
```
Comment (JounQin) [inline esm.mjs:60]: Wondering why this function is marked `deprecated` since `encode` function uses it actually.

File: package.json [hunk 2]
```diff
@@ -4,17 +4,16 @@
   "description": "Yet another Base64 transcoder in pure-JS",
   "main": "base64.js",
   "module": "base64.mjs",
-  "browser": "base64-es5.js",
```
Comment (JounQin) [inline package.json:7]: @dallonf Would you like to publish a patch version? The incorrect `browser` field may break some user cases of bundlers.

File: esm.mjs (hunk 1 above)
Comment (dankogai) [reply esm.mjs:60]: Yes it is used internally.  It should be not used externally but I was exposed for for some reason.

**Comment by dankogai** [comment]
Thank you so much! now `js-base64` is not just ES-module compatible, it is ES-module savvy.   Just one thing: I wanted to keep both `base64.js` and `base64.mjs` monolithic so all you need is a single script file.

```javascript
export * as Base64 from './esm.mjs;
```

is neat but it breaks the monolith.  So I made it monolithic back again in 912053cd719fa94150e29d92d7a9dc87832a571e .

**Comment by JounQin** [comment]
@dankogai That's fine, but now `commonjs` and `esm`'s API are different.

```js
const {decode, encode} = require('js-base64') // not working
const Base64 = require('js-base64') // not working
const {Base64} = require('js-base64') // working

import {decode, encode} from 'js-base64' // working
import * as Base64 from 'js-base64' // working
import {Base64} from 'js-base64' // working
```

I'd prefer same API between `commonjs` and `esm`.

Of course, it could be very easy to fix:

```js
gBase64.Base64 = gBase64;
module.exports = gBase64;
```

Shall I raise another PR to implement it?

File: package.json (hunk 2 above)
Comment (dallonf) [reply package.json:7]: Did you mean to tag @dankogai?

**Pull request merged**